pywarp10/gts.py

```diff
@@ -46,167 +46,55 @@ def is_gts(x: Any) -> bool:
-            self.data = pd.DataFrame(
-                {
-                    "timestamps": data["timestamps"],
-                    "values": data["values"],
-                }
-            )
-        self.classname = data["classname"]
-        self.labels = data["labels"]
-        if "attributes" in data.keys():
-            self.attributes = data["attributes"]
-        # Convert to timestamps only if higest timestamp is greater than 1 day after the epoch
-        if self.data is not None and max(self.data["timestamps"]) > 86400000000:
-            self.data["timestamps"] = pd.to_datetime(self.data["timestamps"], unit="us")
-
-    def to_pandas(self) -> pd.DataFrame:
-        """Convert the GTS to a pandas DataFrame.
-
-        Returns:
-            A pandas DataFrame.
-        """
-        if self.data is not None:
-            df = self.data.copy()
-            n = len(df)
-        else:
-            df = pd.DataFrame({"classname": [self.classname]})
-            n = 1
-        df["classname"] = self.classname
-        for label in self.labels:
-            df[label] = self.labels[label]
-        return df
-
-    def __str__(self) -> str:
-        return self.__repr__()
-
-    def __repr__(self) -> str:
-        if self.classname:
-            name = f"name      : {self.classname}\n"
-        else:
-            name = ""
-        if self.labels:
-            labels = "labels    :\n"
-            for key, value in self.labels.items():
-                labels += f"  {key}={value}\n"
-        else:
-            labels = ""
-        if self.attributes:
-            attributes = "attributes:\n"
-            for key, value in self.attributes.items():
-                attributes += f"  {key}={value}\n"
-        else:
-            attributes = ""
-        if "timestamps" in self.data.columns:
-            data = self.data[["timestamps", "values"]].__repr__()
-            return f"{name}{labels}{attributes}\n\n{data}"
-        else:
-            return f"{name}{labels}{attributes}\n\nEmpty GTS"
-
-    def __eq__(self, other: "GTS") -> bool:
-        """Check if two GTS are equal.
-
-        Args:
-            other: The other GTS to compare with.
-
-        Returns:
-            True if the two GTS are equal.
-        """
-        if isinstance(other, GTS):
-            if (
-                self.classname == other.classname
-                and self.labels == other.labels
-                and self.attributes == other.attributes
-                and self.data.equals(other.data)
-            ):
-                return True
-        return False
+class GTS(pd.DataFrame):
+    def __init__(self, data: Union[Dict, List]):
+        if isinstance(data, dict):
+            data = self._init_gts(data)
+        elif isinstance(data, list):
+            data = self._init_lgts(data)
+        # Convert to timestamps only if higest timestamp is greater than 1 day after the
+        # epoch
+        if len(data.index) > 0 and max(data.index) > 86400000000:
```

Comment:
why the condition?

Reply:
See comment above

GTS in warp10 are all time series with a timestamp which is the number of µs since the epoch.
Conveniently, some functions exists to transform a timestamp into a human readable datetime.

In python, those timestamp can be stored as DateTime objects which keeps the information of
the timestamp but display a human readable datetime.

However, when creating a time serie, we might use a timeserie with timestamp playing the role
of an index (timestamp = [0, 1, 2, 3, ...]). Without this condition, those indexes would be
displayed as [1970-01-01 00:00:00.000000, 1970-01-01 00:00:00.000001, 
1970-01-01 00:00:00.000002, ...] which looses all the interest.

Therefore, I consider that if the highest timestamps is below 1970-01-01 23:59:59, displaying
an index (raw timestamp) is more adequat.